compiler/src/lowering/mappers/functions.rs

```diff
@@ -129,14 +129,15 @@ fn aug_assign(
     op: Node<fe::BinOperator>,
     value: Node<fe::Expr>,
 ) -> Vec<fe::FuncStmt> {
-    let lowered_target = expressions::expr(context, target);
     let original_value_span = value.span;
-    let lowered_value = expressions::expr(context, value);
+    let lowered_target = expressions::expr(context, target.clone());
+    let lowered_lh_value = expressions::expr(context, target);
+    let lowered_rh_value = expressions::expr(context, value);
 
     let new_value_kind = fe::Expr::BinOperation {
-        left: Box::new(lowered_target.clone().new_id()),
```

Comment:
Copying the lowered expression created two separate AST branches with the same nested IDs, which resulted in us calling `add_expression` twice for the same ID. It wasn't necessarily causing a problem, but may have in more complex scenarios.

We now copy the target expression and lower it twice so we have different IDs.